File: app/src/main/java/com/example/myfakequotesapp/Model/QuoteDao.java
package com.example.myfakequotesapp.Model;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface QuoteDao {
    @Query("SELECT * FROM quote")
    LiveData<List<Quote>> getAllQuote();

    @Query("SELECT * FROM quote WHERE id == :quoteId")
    Quote getQuoteByID(int quoteId);

    @Insert
    void insertQuote(Quote quote);

    @Delete
    void DeleteQuote(Quote quote);

    @Update
    void updateQuote(Quote quote);
}

File: app/src/main/java/com/example/myfakequotesapp/Model/QuoteDao.kt
package com.example.myfakequotesapp.Model

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

@Dao
interface QuoteDao {
    @get:Query("SELECT * FROM quote")
    val allQuote: LiveData<List<Quote?>?>?

    @Query("SELECT * FROM quote WHERE id == :quoteId")
    fun getQuoteByID(quoteId: Int): Quote?

    @Insert
    fun insertQuote(quote: Quote?)

    @Delete
    fun DeleteQuote(quote: Quote?)

    @Update
    fun updateQuote(quote: Quote?)
}